src/indexApi.ts

```diff
@@ -36,28 +43,39 @@ async function main(): Promise<void> {
   app.use(cors())
 
   app.get(`/v1/analytics/`, async function(req, res) {
-    const timeperiod: string = req.query.timePeriod.toLowerCase()
-    const start = parseInt(req.query.start)
-    const end = parseInt(req.query.end)
+    let start: string, end: string, pluginId: string, timePeriod: string
+    try {
+      const analyticsQuery = asAnalyticsReq(req.query)
+      start = analyticsQuery.start
+      end = analyticsQuery.end
+      pluginId = analyticsQuery.pluginId
+      timePeriod = analyticsQuery.timePeriod
+    } catch {
+      res.status(400).send(`Missing Request Fields`)
+      return
+    }
+
```

Comment:
const {start, end, pluginID, timePeriod } = analytics query, declare analyticsQuery above the trycatch